internal/telemetry/telemetry.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetry
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path"
+
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/spf13/afero"
+)
+
+const (
+	dirPermissions   = 0700
+	filePermissions  = 0600
+	cacheFilename    = "telemetry"
+	maxCacheFileSize = 100 * 1024 * 1024 // 100MB
```

Comment:
```suggestion
	maxCacheFileSize = 100_000_000 // 100MB
```
https://en.wikipedia.org/wiki/Binary_prefix#Disk_drives

Reply:
[q] How was 100MB chosen here? is there a guideline somewhere or is it our assumption?

will this ever be configurable?

Reply:
It's an arbitrary choice (I was initially thinking of 1GB, Colm suggested 100MB).
It's not something we want to expose to users (eg: via configuration)
The only telemetry related functionality that we will expose to users is the ability to enable/disable telemetry.
Issues encountered with telemetry should rather be logged to file (part of this epic is to add logging).
Otherwise, telemetry should work (and fail) silently in the background without requiring any user interaction.

The purpose of this maximum cache size is merely to ensure that if we are writing to the user's local filesystem, that we don't end up taking up excessive space - users will not be happy with us if our CLIs fill up their hard drive. But that should be a rare occurrence, because event caching is meant for initial (pre-profile) events. As soon as a profile is created events will instead be sent to Atlas.